File: app/src/main/java/com/test/xyz/demo/presentation/weather/presenter/WeatherView.java
package com.test.xyz.demo.presentation.weather.presenter;

public interface WeatherView {
    String getUserNameText();
    String getCityText();

    void showBusyIndicator();
    void hideBusyIndicator();

    void showResult(String result);

    void showUserNameError(int messageId);
    void showCityNameError(int messageId);
    void showGenericError(int messageId);
}

File: app/src/main/kotlin/com/test/xyz/demo/presentation/weather/presenter/WeatherView.kt
package com.test.xyz.demo.presentation.weather.presenter

interface WeatherView {
    fun getUserNameText(): String
    fun getCityText(): String

    fun showBusyIndicator()
    fun hideBusyIndicator()

    fun showResult(result: String)

    fun showUserNameError(messageId: Int)
    fun showCityNameError(messageId: Int)
    fun showGenericError(messageId: Int)
}
